packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -303,6 +294,24 @@ export const PropertyGrid = ({
           }
         }
 
+        // Do any overrides on default menu options
+        if(defaultContextMenuOptions?.size && defaultContextMenuOptions.size > 0 ) {
+          for(const key of Object.values(PropertyGridDefaultContextMenuKey)) {          
+            const overrides = defaultContextMenuOptions?.get(key);
+            const itemIndex = items.map((item) => item.key).indexOf(key);
```

Comment:
you should gate the next 2 lines after getting overrides so that they only run if overrides exist